### PR TITLE
Input/output node shape: replace dynamic axis with -1

### LINKED ISSUE #17: Output node shape: replace 'time' with -1
As the following simple example shows we currently only replace 'time' in the shape of input nodes, but should do the same with output nodes:

```python
import torch
import audonnx


class TorchModel(torch.nn.Module):

    def __init__(
        self,
    ):
        super().__init__()

    def forward(self, x: torch.Tensor):
        return x


torch_model = TorchModel()

input = torch.randn(1, 100, 4)  # batch, time, feature
output = torch_model(input)

onnx_path = 'model.onnx'
torch.onnx.export(
    torch_model,
    input,
    onnx_path,
    input_names=['input'],
    output_names=['output'],
    dynamic_axes={
        'input': {1: 'time'},
        'output': {1: 'time'},
    },
    opset_version=12,
)

audonnx.Model(onnx_path)
```
```
Input:
  output:
    shape: [1, -1, 4]
    dtype: tensor(float)
    transform: None
Output:
  output:
    shape: [1, time, 4]   # should be [1, -1, 4]
    dtype: tensor(float)
    labels: [output-0, output-1, output-2, output-3]
```

### PR DESCRIPTION
**Pull request by frankenjoe**
Closes #17 

In ONNX format axis with a dynamic size are represented by a string. This is not convenient as we cannot directly use it to e.g. reshape an input array to match the expected shape. We therefore replace the string with -1 in the node classes of `audonnx`.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -130,6 +130,34 @@ def forward(self, signal: torch.Tensor, spectrogram: torch.Tensor):
...
+torch.onnx.export(
+    TorchModelDynamic(),
+    torch.randn(pytest.FEATURE_SHAPE),
+    pytest.MODEL_DYNAMIC_PATH,
+    input_names=['input'],
+    output_names=['output'],
+    dynamic_axes={
+        'input': {1: 'time'},
+        'output': {1: 'time'},
+    },
```
Comment (hagenw) [inline tests/conftest.py:156]: What happens if I export using

```python
    dynamic_axes={
        'input': {1: 'magnitude'},
        'output': {1: 'magnitude'},
    },
```

Comment (frankenjoe) [reply tests/conftest.py]: Changed it accordingly and updated the description.

File: audonnx/core/model.py [hunk 2]
```diff
@@ -304,3 +306,10 @@ def _device_to_providers(
     else:
         providers = device
     return providers
+
+
+def _shape(
+        shape: typing.List[typing.Union[int, str]],
+) -> typing.List[int]:
+    r"""Replace 'time' with -1."""
```
Comment (hagenw) [inline audonnx/core/model.py:314]:
```suggestion
    r"""Replace dynamic axis names with -1."""
```
